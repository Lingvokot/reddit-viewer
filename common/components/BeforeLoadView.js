import React, { Component } from "react";
import { Button, View, Text } from "react-native";

export default class BeforeLoadView extends Component {
	render() {
		if (!this.props.finished)
			return (
				<Text>Loading posts, please wait...</Text>
			);
		else if (this.props.error)
			return (
				<View>
					<Text>{this.props.error}</Text>
					<Button title="Retry to load"
							onPress={this.props.retryFunc}
							style={{
								paddingTop: 5, paddingBottom: 5,
								margin: 5
							}} ref={(ref) => {this.retryButton = ref;}}/>
				</View>
			);
		else
			return null;
	}
}