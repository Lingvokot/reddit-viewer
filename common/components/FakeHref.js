import React, { Component, PropTypes } from "react";
import { TouchableHighlight, Linking, Text } from "react-native";

export default class FakeHref extends Component {
	render() {
		return (
			<TouchableHighlight
				onPress={() => {
					return Linking.canOpenURL(this.props.url).then(supported => {
						if (!supported)
							this.props.onCannotHandle();
						else
							return Linking.openURL(this.props.url);
					}).catch(this.props.errorHandler);
				}} ref={(ref) => {this.highlight = ref;}}>
				<Text style={{color: "blue", textDecorationLine: "underline"}}>
					{this.props.url}
				</Text>
			</TouchableHighlight>
		);
	}
}
