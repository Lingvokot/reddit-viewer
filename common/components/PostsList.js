import React, { Component, PropTypes } from 'react';
import { View, Text, TouchableHighlight, ListView } from 'react-native';

export default class PostsList extends Component {
	render() {
		this.listElements = [];
		if (this.props.toList.getRowCount() == 0)
			return (
				<Text>There are no posts yet.</Text>
			)
		return (
			<ListView dataSource={this.props.toList}
				onEndReached={() => {
					this.props.onEndReached();
				}}
				renderRow={(rowData) => (
					<TouchableHighlight style={{
							borderWidth: 1,
							borderRadius: 3,
							borderColor: "red",
							margin: 5,
							padding: 5
						}} ref={(ref) => {this.listElements.push(ref);}}
						onPress={() => {this.props.onForward(rowData);}}>
						<Text>{rowData.title}</Text>
					</TouchableHighlight>
				)} ref={(ref) => {this.listView = ref;}}/>
		);
	}
}