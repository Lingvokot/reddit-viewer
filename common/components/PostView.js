/*global alert*/
import React, { Component, PropTypes } from "react";
import { View, Text, Button, Modal } from "react-native";
import FakeHref from "./FakeHref";

export default class PostView extends Component {
	constructor(props) {
		super(props);
		this.state = {urlOpenError: null};
	}
	render() {
		let selectedPost = this.props.selectedPost;
		return (
			<View>
				<Button onPress={this.props.onBack}
						title="Back"
						style={{
							paddingTop: 5, paddingBottom: 5,
							backgroundColor: "blue"
						}} ref={(ref) => {this.backButton = ref;}}/>
				<Text style={{fontWeight: "bold", textAlign: "center"}}>
					{selectedPost.title}
				</Text>
				<Text>{`ID: ${selectedPost.name}`}</Text>
				<Text>{`Author: ${selectedPost.author}`}</Text>
				<Text>{`Score: ${selectedPost.score}`}</Text>
				<Text>External URL:</Text>
				<FakeHref url={selectedPost.cleanUrl}
					onCannotHandle={() => {
						alert("Cannot handle this URL");
					}}
					errorHandler={(err) => {
						alert(err.message);
					}} ref={(ref) => {this.externalLink = ref;}}/>
			</View>
		);
	}
}