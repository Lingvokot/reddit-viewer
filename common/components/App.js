import React, { Component } from 'react';
import { Navigator, ListView } from 'react-native';
import PostsList from './PostsList';
import PostView from './PostView';
import BeforeLoadView from './BeforeLoadView';
import { last } from 'lodash/array';
import { each } from 'lodash/collection';
import loadPosts from './loadPosts';

const cardsInPage = 10;

export default class App extends Component {
	constructor(props) {
		super(props);
		this.state = {
			selectedPost: null, postsToList: [], pagesFetched: 0,
			error: null, finished: false, inProcess: false
		};
	}
	componentWillMount() {
		if (this.state.postsToList.length == 0)
			this.loadMorePosts();
	}
	loadMorePosts() {
		if (this.state.inProcess)
			return;
		return new Promise((resolve) => {
			this.setState({finished: false, inProcess: true}, async () => {
				let page = null;
				try {
					if (this.state.postsToList.length == 0)
						page = await loadPosts({limit: 10});
					else {
						const after = last(this.state.postsToList).uuid;
						page = await loadPosts({after, limit: 10});
					}
				}
				catch (e) {
					this.setState({error: e.message, finished: true, inProcess: false}, () => resolve());
					return;
				}
				let newPosts = Object.values(page.apiResponse.posts);
				this.setState({
					error: null, finished: true, inProcess: false,
					postsToList: this.state.postsToList.concat(newPosts),
					pagesFetched: this.state.pagesFetched + 1
				}, () => resolve());
			});
		});
	}
	render() {
		if (this.state.error || (!this.state.finished && this.state.postsToList.length == 0))
			return (
				<BeforeLoadView error={this.state.error}
					finished={this.state.finished}
					retryFunc={() => this.loadMorePosts()}/>
			)
		return (
			<Navigator initialRoute={{ title: 'Posts', index: 0 }}
				renderScene={(route, navigator) => {
				const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
				if (route.index == 0)
					return (
						<PostsList toList={ds.cloneWithRows(this.state.postsToList)}
							onForward={(selectedPost) => {
								return new Promise(resolve => {
									this.setState({selectedPost});
									resolve();
									navigator.push({title: selectedPost.title, index: route.index + 1});
								}).catch(err => {});
							}}
							onEndReached={() => this.loadMorePosts()}
							ref={(ref) => {this.postsList = ref;}}/>
					)
				else
					return (
						<PostView selectedPost={this.state.selectedPost}
							onBack={() => {
								return new Promise(resolve => {
									this.setState({selectedPost: null});
									resolve();
									navigator.pop();
								}).catch(err => {});
							}} ref={(ref) => {this.postView = ref;}}/>
					)
			}} />
		);
	}
}