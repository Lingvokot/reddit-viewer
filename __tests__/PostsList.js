/*global jest, describe, it, test, jasmine, expect, console*/
/*eslint no-console: "off"*/
import "react-native";
import {ListView, Linking} from "react-native";
import React from "react";
import Index from "../index.android.js";
import App from "../common/components/App";
import BeforeLoadView from "../common/components/BeforeLoadView";
import PostView from "../common/components/PostView";
import PostsList from "../common/components/PostsList";
import FakeHref from "../common/components/FakeHref";
const {configureMock, makeArtificialErrorEvent} = Linking;

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
jest.useFakeTimers();
jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;

const answer2 = {
	"posts": {
		"t3_5vpnjl": {
			"cleanUrl":"http://www.businessinsider.com.au/elijah-wood-hollywood-pedophile-ring-2016-5?r=US&IR=T",
			"createdUTC":1487844877,"name":"t3_5vpnjl","author":"yeoldgolddigger",
			"title":"Elijah Wood says there's a secret paedophile ring in Hollywood and child actors are 'prey'",
			"score":4329
		},
		"t3_5vo7nw": {
			"cleanUrl":"https://i.redd.it/9un6j34abjhy.png",
			"createdUTC":1487824111,"name":"t3_5vo7nw","author":"iwvp",
			"title":"Alien visits /pol/",
			"score":16119
		},
		"t3_5vo5en": {
			"cleanUrl":"https://motherboard.vice.com/en_us/article/3000-scientists-have-asked-for-help-running-for-office-to-oppose-trump",
			"createdUTC":1487823483,"name":"t3_5vo5en","author":"200357931",
			"title":"3,000 Scientists Have Asked for Help Running for Office to Oppose Trump",
			"score":13851
		},
		"t3_5vol5q": {
			"cleanUrl":"http://www.cnn.com/2017/02/22/politics/boy-toby-question-tom-cotton-town-hall-trnd/",
			"createdUTC":1487828174,"name":"t3_5vol5q","author":"doogie92",
			"title":"7-year-old to senator: Don't take away PBS to build the wall.",
			"score":6880
		},
		"t3_5vo599": {
			"cleanUrl":"https://www.nytimes.com/2015/01/14/opinion/tom-hanks-on-his-two-years-at-chabot-college.html?_r=1",
			"createdUTC":1487823444, "name":"t3_5vo599", "author":"freddyjohnson",
			"title":"TIL Tom Hanks is a big fan of community colleges. Since he had poor grades in high school and lousy SAT scores Hanks ended up at Chabot, a nearby community college, because it accepted everyone and was free.",
			"score":12347
		},
		"t3_5vo8c6": {
			"cleanUrl":"http://i.imgur.com/lxVpCqu.gifv",
			"createdUTC":1487824312,"name":"t3_5vo8c6","author":"ClosingDownSummer",
			"title":"Just a guy taking down the Confederate Flag on live TV",
			"score":38763
		},
		"t3_5vo1ow": {
			"cleanUrl":"/r/Showerthoughts/comments/5vo1ow/as_a_child_i_pretended_to_like_carrots_for_my/",
			"createdUTC":1487822460, "name":"t3_5vo1ow", "author":"Jalexgeorge",
			"title":"As a child I pretended to like carrots for my parents. As an adult I pretend to like them for my child. I fucking hate carrots.",
			"score":23182
		},
		"t3_5voqhq": {
			"cleanUrl":"https://i.redd.it/yyosucwfsjhy.gif",
			"createdUTC":1487829864,"name":"t3_5voqhq","author":"nyr3188",
			"title":"Runaway truck",
			"score":4727
		},
		"t3_5vnlu2": {
			"cleanUrl":"https://www.theguardian.com/world/2017/feb/23/north-korea-malaysia-investigating-kim-jong-nam-death",
			"createdUTC":1487818229,"name":"t3_5vnlu2","author":"EpycWyn",
			"title":"North Korea demands Malaysia stop investigating Kim Jong-nam death",
			"score":20560
		},
		"t3_5vnp96": {
			"cleanUrl":"http://i.imgur.com/GPxQTZY.jpg",
			"createdUTC":1487819173,"name":"t3_5vnp96","author":"Ferenginar",
			"title":"$0.25 out of a machine. $947.36 out of a dog. Gotta love that smug \"worth it\" look.",
			"score":48443
		},

	}
};

describe("Testing PostsList", () => {
  let returnedValue = null;
	const content = Object.values(answer2.posts);
  const mockOnForward = jest.fn();
	mockOnForward.mockImplementation((row) => {returnedValue = row;});
  const mockOnEndReached = jest.fn();
	mockOnEndReached.mockImplementation(() => {});
  const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});

  test("Rendering case: there are no posts yet", () => {
		const domTree = renderer.create(
  		<PostsList toList={ds.cloneWithRows([])} onForward={mockOnForward}
        onEndReached={mockOnEndReached}/>
  	);
  	const tree = domTree.toJSON();
  	expect(tree).toMatchSnapshot();
	});
  test("Rendering case: there are some posts", () => {
  	const domTree = renderer.create(
  		<PostsList toList={ds.cloneWithRows(content)}
				onForward={mockOnForward} onEndReached={mockOnEndReached}/>
  	);
  	const tree = domTree.toJSON();
		expect(tree).toMatchSnapshot();
  });
	test("Checking reaction on onEndReached event", () => {
  	const domTree = renderer.create(
  		<PostsList toList={ds.cloneWithRows(content)} onForward={mockOnForward}
        onEndReached={mockOnEndReached}/>
  	);
  	const listView = domTree.getInstance().listView;
  	expect(typeof listView.props.onEndReached).toEqual("function");
		listView.props.onEndReached();
  	expect(mockOnEndReached).toBeCalled();
  });
	test("Checking touchability of post headers", () => {
  	const domTree = renderer.create(
  		<PostsList toList={ds.cloneWithRows(content)}
				onForward={mockOnForward} onEndReached={mockOnEndReached}/>
  	);
  	const listElements = domTree.getInstance().listElements;
		const expectedLength = Object.keys(answer2.posts).length;
  	expect(listElements.length).toEqual(expectedLength);
  	for (let i = 0; i < expectedLength; i++) {
			const element1 = listElements[i];
  		expect(typeof element1.props.onPress).toEqual("function");
  		element1.props.onPress();
  		expect(mockOnForward.mock.calls.length).toEqual(i + 1);
			expect(JSON.stringify(returnedValue)).toEqual(JSON.stringify(content[i]));
  	}
  });
});
