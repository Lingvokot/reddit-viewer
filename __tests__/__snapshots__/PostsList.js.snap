// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PostsList Rendering case: there are no posts yet 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  There are no posts yet.
</Text>
`;

exports[`Testing PostsList Rendering case: there are some posts 1`] = `
<View>
  <RCTScrollView>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderColor": "red",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Elijah Wood says there's a secret paedophile ring in Hollywood and child actors are 'prey'
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderColor": "red",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Alien visits /pol/
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderColor": "red",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          3,000 Scientists Have Asked for Help Running for Office to Oppose Trump
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderColor": "red",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          7-year-old to senator: Don't take away PBS to build the wall.
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderColor": "red",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          TIL Tom Hanks is a big fan of community colleges. Since he had poor grades in high school and lousy SAT scores Hanks ended up at Chabot, a nearby community college, because it accepted everyone and was free.
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderColor": "red",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Just a guy taking down the Confederate Flag on live TV
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderColor": "red",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          As a child I pretended to like carrots for my parents. As an adult I pretend to like them for my child. I fucking hate carrots.
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderColor": "red",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Runaway truck
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderColor": "red",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          North Korea demands Malaysia stop investigating Kim Jong-nam death
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "borderColor": "red",
              "borderRadius": 3,
              "borderWidth": 1,
              "margin": 5,
              "padding": 5,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          $0.25 out of a machine. $947.36 out of a dog. Gotta love that smug "worth it" look.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
